weather_test.go

```diff
@@ -1,61 +1,66 @@
 package weather_test
 
 import (
-	"bufio"
-	"fmt"
+	"encoding/json"
+	"io"
 	"net/http"
 	"net/http/httptest"
 	"os"
 	"testing"
 	"weather"
 )
 
-func GetTestCases() []string {
-
-	file, err := os.Open("testcases.txt")
-	if err != nil {
-		fmt.Printf("%v", err)
-	}
-	defer file.Close()
-
-	var lines []string
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		lines = append(lines, scanner.Text())
-	}
-	return lines
+type Response struct {
+	OneWord string  `json:"oneword"`
+	Celcius float64 `json:"celcius"`
+	City    string  `json:"city"`
 }
 
-func WeatherNewTLSServer(testcases string) (r http.Response) {
+func TestGetWeather(t *testing.T) {
+
+	var r Response
 
 	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, testcases)
+		file, err := os.Open("testdata/weather.json")
+		if err != nil {
+			t.Fatal(err)
+		}
+		io.Copy(w, file)
 	}))
 	defer ts.Close()
 
 	client := ts.Client()
 	res, err := client.Get(ts.URL)
 	if err != nil {
-		fmt.Printf("%v", err)
+		t.Fatal(err)
 	}
 
-	return *res
-}
+	want := weather.Conditions{
+		City:    "Birmingham",
+		OneWord: "Clouds",
+		Celcius: 23,
+	}
 
-func TestGetWeather(t *testing.T) {
+	got, err := weather.Get(*res)
 
-	// var conditions []byte
+	if err != nil {
+		t.Fatal(err)
+	}
 
-	testcases := GetTestCases()
-	for _, testcase := range testcases {
-		resp := WeatherNewTLSServer(testcase)
-		conditions, err := weather.Get(resp)
+	json.Unmarshal(got, &r)
+	OneWordResponse := &r.OneWord
+	CityResponse := &r.City
+	CelciusResonse := &r.Celcius
 
-		if err != nil {
-			t.Fatal(err)
-		}
-		if len(conditions) == 0 {
-			t.Fatal("no conditions")
-		}
+	if want.OneWord != *OneWordResponse {
```

Comment:
Instead of comparing each struct field, one by one, we could compare the entire struct, couldn't we?